### PR TITLE
Add sphinx.util.typing:unicode to help mypy-3 migration

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Refactoring

### Purpose
- To accelerate migration to mypy-py3 mode, I'd like to introduce `sphinx.util.typing:unicode` type.
- It allows to switch naturally `unicode` to `str` by `python_version` option for mypy.
- It will be removed after all migration finished.
- @cocoatomo Do you have any opinion? Is this helpful for you (and me)?

### CODE REVIEW

File: sphinx/addnodes.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 if False:
     # For type annotation
     from typing import List, Sequence  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/addnodes.py]: This is wrong modname...

File: sphinx/deprecation.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
     # For type annotation
     # note: Don't use typing.TYPE_CHECK here (for py27 and py34).
     from typing import Any, Dict, Type  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/deprecation.py]: This is also typo

File: sphinx/events.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
 if False:
     # For type annotation
     from typing import Any, Callable, Dict, List  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/events.py]: typo

File: sphinx/locale/__init__.py [hunk 4]
```diff
@@ -23,6 +23,7 @@
 if False:
     # For type annotation
     from typing import Any, Callable, Dict, Iterator, List, Tuple  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/locale/__init__.py]: typo

File: sphinx/pycode/parser.py [hunk 5]
```diff
@@ -22,6 +22,7 @@
 if False:
     # For type annotation
     from typing import Any, Dict, IO, List, Tuple  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/pycode/parser.py]: typo

File: sphinx/testing/path.py [hunk 6]
```diff
@@ -15,6 +15,7 @@
 if False:
     # For type annotation
     from typing import Any, Callable, IO, List  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/testing/path.py]: typo

File: sphinx/util/console.py [hunk 7]
```diff
@@ -22,6 +22,7 @@
 if False:
     # For type annotation
     from typing import Dict  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/console.py]: typo

File: sphinx/util/docstrings.py [hunk 8]
```diff
@@ -14,6 +14,7 @@
 if False:
     # For type annotation
     from typing import List  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/docstrings.py]: typo

File: sphinx/util/images.py [hunk 9]
```diff
@@ -34,6 +34,7 @@
 if False:
     # For type annotation
     from typing import Dict, IO, List, Tuple  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/images.py]: typo

File: sphinx/util/jsdump.py [hunk 10]
```diff
@@ -17,6 +17,7 @@
 if False:
     # For type annotation
     from typing import Any, Dict, IO, List, Match, Union  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/jsdump.py]: typo

File: sphinx/util/jsonimpl.py [hunk 11]
```diff
@@ -17,6 +17,7 @@
 if False:
     # For type annotation
     from typing import Any, IO  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/jsonimpl.py]: typo

File: sphinx/util/matching.py [hunk 12]
```diff
@@ -14,6 +14,7 @@
 if False:
     # For type annotation
     from typing import Callable, Dict, List, Match, Pattern  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/matching.py]: typo

File: sphinx/util/osutil.py [hunk 13]
```diff
@@ -29,6 +29,7 @@
 if False:
     # For type annotation
     from typing import Any, Iterator, List, Tuple, Union  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/osutil.py]: typo

File: sphinx/util/parallel.py [hunk 14]
```diff
@@ -26,6 +26,7 @@
 if False:
     # For type annotation
     from typing import Any, Callable, Dict, List, Sequence  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/parallel.py]: typo

File: sphinx/util/png.py [hunk 15]
```diff
@@ -12,6 +12,10 @@
 import binascii
 import struct
 
+if False:
+    # For type annotation
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/png.py]: typo

File: sphinx/util/pycompat.py [hunk 16]
```diff
@@ -22,6 +22,7 @@
 if False:
     # For type annotation
     from typing import Any, Callable, Generator  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/pycompat.py]: typo

File: sphinx/util/rst.py [hunk 17]
```diff
@@ -23,6 +23,7 @@
 if False:
     # For type annotation
     from typing import Generator  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/rst.py]: typo

File: sphinx/util/stemmer/__init__.py [hunk 18]
```diff
@@ -11,6 +11,10 @@
 
 from sphinx.util.stemmer.porter import PorterStemmer
 
+if False:
+    # For type annotation
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/stemmer/__init__.py]: typo

File: sphinx/util/stemmer/porter.py [hunk 19]
```diff
@@ -28,6 +28,10 @@
     :license: Public Domain ("can be used free of charge for any purpose").
 """
 
+if False:
+    # For type annotation
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/stemmer/porter.py]: typo

File: sphinx/util/tags.py [hunk 20]
```diff
@@ -17,6 +17,7 @@
 if False:
     # For type annotation
     from typing import Iterator, List  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/tags.py]: typo

File: sphinx/util/texescape.py [hunk 21]
```diff
@@ -14,6 +14,7 @@
 if False:
     # For type annotation
     from typing import Dict  # NOQA
+    from sphinx.typing import unicode  # NOQA
```
Comment (tk0miya) [inline sphinx/util/texescape.py]: typo

**Review by tk0miya — COMMENTED** [review]
OMG, too many typos...

**Comment by cocoatomo** [comment]
LGTM 👍

**Pull request merged**